code/modules/mining/mint.dm

```diff
@@ -1,198 +1,245 @@
+
+	dirlist[changingdir] = newdir
+
+	in_dir = dirlist[1]
+	out_dir = dirlist[2]
+	return 1
+
+/obj/machinery/mineral/mint/proc/DropSheet(var/matID)
+	var/datum/material/M = materials.getMaterial(matID)
+	var/obj/item/stack/sheet/sh = new M.sheettype(src.loc)
+	if(sh)
+		var/available_num_sheets = materials.storage[matID]
+		if(available_num_sheets>0)
+			//available_num_sheets % sh.max_amount
+			sh.amount = available_num_sheets
+			materials.removeAmount(matID, sh.amount)
+		else
+			qdel(sh)
+	return 1
 
 /obj/machinery/mineral/mint/Topic(href, href_list)
-	if(..())
+	. = ..()
+	if(.)
 		return
 	usr.set_machine(src)
 	src.add_fingerprint(usr)
+	var/turf/out_T = get_step(src, out_dir)
+
+	if(href_list["close"])
+		usr.unset_machine(src)
+		return 1
+
 	if(processing==1)
 		to_chat(usr, "<span class='notice'>The machine is processing.</span>")
 		return
+
+	if(href_list["eject"])
+		var/datum/material/ma = materials.getMaterial(href_list["eject"])
+		var/obj/item/stack/sheet/sh = new ma.sheettype(out_T)
+		sh.amount = materials.getAmount(href_list["eject"])
+		materials.removeAmount(href_list["eject"], sh.amount)
+		if (chosen == href_list["eject"])
+			chosen = null
+
+	if("changedir" in href_list)
+		//Change_Dir()
+		var/changingdir = text2num(href_list["changedir"]) //See ore processing_unit for original comments
+		changingdir = Clamp(changingdir, 1, 2)
+
+		var/newdir = input("Select the new direction", name, "North") as null|anything in list("North", "South", "East", "West")
+		if(!newdir)
+			return 1
+		newdir = text2dir(newdir)
+
+		var/list/dirlist = list(in_dir, out_dir)
+		var/olddir = dirlist[changingdir]
+		dirlist[changingdir] = -1
+
+		var/conflictingdir = dirlist.Find(newdir)
+		if(conflictingdir)
+			dirlist[conflictingdir] = olddir
+
+		dirlist[changingdir] = newdir
+
+		in_dir = dirlist[1]
+		out_dir = dirlist[2]
+
 	if(href_list["choose"])
 		chosen = href_list["choose"]
+
 	if(href_list["chooseAmt"])
 		coinsToProduce = Clamp(coinsToProduce + text2num(href_list["chooseAmt"]), 0, 1000)
+
 	if(href_list["makeCoins"])
+		if(chosen == null)
+			return
 		var/temp_coins = coinsToProduce
-		if (src.output)
+		if (src.out_dir)
 			processing = 1
 			icon_state = "coinpress1"
-			//var/obj/item/weapon/storage/bag/money/M
 			var/datum/material/po=materials.getMaterial(chosen)
 			if(!po)
 				chosen=null
 				processing=0
 				return
 			while(materials.storage[chosen] > 0 && coinsToProduce > 0)
-			/*	if (locate(/obj/item/weapon/storage/bag/money,output.loc))
-					M = locate(/obj/item/weapon/storage/bag/money,output.loc)
-					if(M.can_be_inserted(po.cointype, 1))
-						new po.cointype(M)
+				var/obj/item/weapon/storage/bag/money/tempbag = locate(/obj/item/weapon/storage/bag/money,out_T)
+				materials.removeAmount(chosen, 1) //We'll get that money up front don't you worry.
+				for(var/i=0,i<coins_per_sheet,i++)
+					var/obj/item/weapon/coin/co = new po.cointype
+					if(tempbag)
+						if(tempbag.can_be_inserted(co, 1))
```

Comment:
So if I'm reading this correctly this will leave the spawned coin in nullspace if there is a bag but it's full?

Reply:
I'm gonna have to blame my git client on this one i corrected that already, my bad for not noticing.
I think it's time for a new client but for now it's time for yet another force push. For what it's worth i made sure to retest it and it works fine as far as the bag limit of 300 is concerned.

Reply:
>300
wew
does that cover up the entire screen?

Reply:
No the bag compresses it to one slot per coin type with a number on the bottom left, the part replacer could do with that as well.